webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala

```diff
@@ -291,14 +302,35 @@ class BinaryDataController @Inject()(
             (dataSource, dataLayer) <- getDataSourceAndDataLayer(organizationName, dataSetName, dataLayerName)
             segmentationLayer <- tryo(dataLayer.asInstanceOf[SegmentationLayer]).toFox ?~> Messages("dataLayer.notFound")
             mappingRequest = DataServiceMappingRequest(dataSource, segmentationLayer, mappingName)
-            result <- binaryDataService.handleMappingRequest(mappingRequest)
+            result <- mappingService.handleMappingRequest(mappingRequest)
           } yield {
             Ok(result)
           }
         }
       }
   }
 
+  /**
+    * Handles isosurface requests.
+    */
+  def requestIsosurface(organizationName: String, dataSetName: String, dataLayerName: String) = Action.async(validateJson[WebKnossosIsosurfaceRequest]) {
+    implicit request =>
+      accessTokenService.validateAccess(UserAccessRequest.readDataSources(DataSourceId(dataSetName, organizationName))) {
+        AllowRemoteOrigin {
+          for {
+            (dataSource, dataLayer) <- getDataSourceAndDataLayer(organizationName, dataSetName, dataLayerName)
+            segmentationLayer <- tryo(dataLayer.asInstanceOf[SegmentationLayer]).toFox ?~> Messages("dataLayer.mustBeSegmentation")
+            isosurfaceRequest = IsosurfaceRequest(dataSource, segmentationLayer, request.body.cuboid(dataLayer), request.body.segmentId, request.body.voxelDimensions, request.body.mapping)
+            vertices <- isosurfaceService.requestIsosurfaceViaActor(isosurfaceRequest)
+          } yield {
+            val responseBuffer = ByteBuffer.allocate(vertices.length * 4).order(ByteOrder.LITTLE_ENDIAN)
+            responseBuffer.asFloatBuffer().put(vertices)
+            Ok(responseBuffer.array())
```

Comment:
please add a comment about the float array semantics (3 floats being a point etc)